opencog/atoms/bind/PatternLink.cc

```diff
@@ -482,9 +475,58 @@ void PatternLink::unbundle_virtual(const std::set<Handle>& vars,
 {
 	for (const Handle& clause: clauses)
 	{
-		bool is_virtual = false;
 		bool is_black = false;
 
+		// Check if all links from top to bottom are either VirtualLink,
+		// logical links, EvaluationLink with GroundedPredicateNode, or
+		// ExecuationOutputLink; cause otherwise none of the evaluatable
+		// can be evaluated and must be matched directly.
+		std::function<bool (const Handle&)> is_virtualable = [&](const Handle& h)
```

Comment:
Not 100% the same (even if it is less general).  The new method check to make sure only clauses that are truly virtual will be marked as is, so if an evaluatable is under a link that accepts atom instead of TV, then the clause is not virtual.